### PR TITLE
tECDSA signing: TSS round 9

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Refs: https://github.com/keep-network/keep-core/issues/3042
Depends on: https://github.com/keep-network/keep-core/pull/3249

Here we implement the TSS round nine for the tECDSA signing protocol. This round expects simple broadcast messages produced in round eight as input and produces a simple broadcast message as output.

### Next steps

This PR is just a part of the entire work regarding the tECDSA signing test loop. Please refer to the description of https://github.com/keep-network/keep-core/issues/3042 for the full roadmap and the current status of the work. At this point, the next step will be introducing the finalization of TSS signing.

### CODE REVIEW

File: pkg/tecdsa/signing/marshaling.go [hunk 1]
```diff
@@ -300,6 +300,34 @@ func (trem *tssRoundEightMessage) Unmarshal(bytes []byte) error {
...
+// network communication.
+func (trnm *tssRoundNineMessage) Marshal() ([]byte, error) {
+	return (&pb.TSSRoundNineMessage{
+		SenderID:  uint32(trnm.senderID),
+		Payload:   trnm.payload,
+		SessionID: trnm.sessionID,
+	}).Marshal()
+}
+
+// Unmarshal converts a byte array produced by Marshal to an tssRoundNineMessage.
```
Comment (tomaszslabon) [inline pkg/tecdsa/signing/marshaling.go:313]: an tssRoundNineMessage -> a tssRoundNineMessage

File: pkg/tecdsa/signing/member.go [hunk 2]
```diff
@@ -357,5 +357,31 @@ func (trem *tssRoundEightMember) markInactiveMembers(
...
+}
+
+// tssRoundNineMember represents one member in a signing group performing the
+// ninth round of the TSS keygen.
+type tssRoundNineMember struct {
+	*tssRoundEightMember
+}
+
+// markInactiveMembers takes all messages from the previous signing protocol
+// execution phase and marks all member who did not send a message as inactive.
```
Comment (tomaszslabon) [inline pkg/tecdsa/signing/member.go:377]: member -> members

**Review by tomaszslabon — APPROVED** [review]

File: pkg/tecdsa/signing/marshaling.go (hunk 1 above)
Comment (lukasz-zimnoch) [reply pkg/tecdsa/signing/marshaling.go:313]: Addressed by https://github.com/keep-network/keep-core/pull/3271

File: pkg/tecdsa/signing/member.go (hunk 2 above)
Comment (lukasz-zimnoch) [reply pkg/tecdsa/signing/member.go:377]: Addressed by https://github.com/keep-network/keep-core/pull/3271

File: pkg/tecdsa/signing/message.go [hunk 3]
```diff
@@ -190,4 +190,24 @@ func (trem *tssRoundEightMessage) SenderID() group.MemberIndex {
...
+	payload   []byte
+	sessionID string
+}
+
+// SenderID returns protocol-level identifier of the message sender.
+func (trnm *tssRoundNineMessage) SenderID() group.MemberIndex {
+	return trnm.senderID
+}
+
+// Type returns a string describing an tssRoundNineMessage type for
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply pkg/tecdsa/signing/message.go:209]: Addressed by https://github.com/keep-network/keep-core/pull/3271

**Pull request merged**